api/envoy/config/core/v3/protocol.proto

```diff
@@ -386,6 +386,11 @@ message Http2ProtocolOptions {
   // Send HTTP/2 PING frames to verify that the connection is still healthy. If the remote peer
   // does not respond within the configured timeout, the connection will be aborted.
   KeepaliveSettings connection_keepalive = 15;
+
+  // `SETTINGS_MAX_HEADER_LIST_SIZE <https://tools.ietf.org/html/rfc7540>`_
+  //  This option informs upstream peers the maximum size of header list the sender is ready to accept
+  //  in octets. Initial value of this setting is unlimited.
```

Comment:
Please add to the comments that requests that exceed this value will receive a 431 response.

Reply:
Done.